### PR TITLE
Adding receipts to expenses

### LINKED ISSUE #331: Add receipts to payments
Users should be able to (or required to) attach photos of the receipts when they upload an expense

### PR DESCRIPTION
**Pull request by viktorcsimma**
Closes #331. Receipts are required for expenses; they can also be downloaded from the list, but they cannot be modified.

### CODE REVIEW

File: resources/views/utils/checkout/add-expense.blade.php [hunk 1]
```diff
@@ -33,6 +33,8 @@
                 <x-input.select m=6 l=6 id="payer" text="Fizető" :elements="\App\Models\User::collegists()" default="{{user()->id}}" :formatter="function($user) { return $user->uniqueName; }" />
                 <x-input.checkbox m=6 l=6 id="paid" checked text="A tartozás kifizetésre került"/>
                 @endcan
+                <x-input.file s="12" xl="8" id="receipt" style="margin-top:auto" accept=".pdf,.jpg,.png,.jpeg,.gif"
```
Comment (kdmnk) [inline resources/views/utils/checkout/add-expense.blade.php]: Only one question: why xl=8? I'd just delete it.

File: app/Utils/CheckoutHandler.php [hunk 2]
```diff
@@ -209,6 +212,9 @@ public function addExpense(Request $request): RedirectResponse
             'paid_at'           => $paid ? Carbon::now() : null,
         ]);
 
+        $path = $request->file('receipt')->store('receipts');
```
Comment (kdmnk) [inline app/Utils/CheckoutHandler.php]: Wrap these and the Transaction::create in a DB::transaction(...), so if either of the logic fails it backtracks. Same with deletion.

**Review by kdmnk — CHANGES_REQUESTED** [review]

**Comment by viktorcsimma** [comment]
Done:)

**Review by kdmnk — APPROVED** [review]
Codewise it's fine, you can test it live :P

One comment though: you can put the mail sending into the transaction as well. Now if that fails, the user gets an error message, yet the transactions are created properly. Depends how we want it.

**Comment by viktorcsimma** [comment]
Hm... I have arguments both for and against it. Such a situation would be confusing; but then if anything happens with the mail function, it does not paralyse the function itself.

I think I'm going to merge it now; thank you:)

**Pull request merged**